src/modules/auth/auth.service.ts

```diff
@@ -1,126 +1,166 @@
+import { UserEntity } from './entities/user.entity';
 import { JwtPayload } from '../../types/jwt-payload.interface';
 
-/**
- * Authentication service
- * Provides JWT token management and authentication infrastructure
- * for all authentication methods (OAuth, Magic Link, etc.)
- */
 @Injectable()
 export class AuthService {
-  private readonly SALT_ROUNDS = 10;
+  private readonly logger = new Logger(AuthService.name);
 
-  constructor(
+  public constructor(
+    private readonly configService: ConfigService,
     @InjectRepository(UserEntity)
     private readonly userRepo: Repository<UserEntity>,
-    private readonly jwtService: JwtService,
-    private readonly configService: ConfigService,
+    @InjectRepository(MagicLinkTokenEntity)
+    private readonly tokenRepo: Repository<MagicLinkTokenEntity>,
+    private readonly dataSource: DataSource,
+    private readonly emailService: EmailService,
+    private readonly jwtService: NestJwtService,
   ) {}
 
-  /**
-   * Register a new user (TEMPORARY - for testing JWT infrastructure)
-   * TODO: Remove in production - users will authenticate via OAuth/Magic Link
-   * @param loginDto - User credentials
-   * @returns Authentication response with JWT tokens
-   * @throws ConflictException if username already exists
-   */
-  public async register(loginDto: LoginDto): Promise<AuthResponseDto> {
-    const existingUser = await this.userRepo.findOne({
-      where: { username: loginDto.username },
-    });
-
-    if (existingUser) {
-      throw new ConflictException('Username already exists');
-    }
+  private mapToUserResponse(user: UserEntity): UserResponseDto {
+    const dto = new UserResponseDto();
+    dto.id = user.id;
+    dto.email = user.email;
+    return dto;
+  }
 
-    const hashedPassword = await bcrypt.hash(
-      loginDto.password,
-      this.SALT_ROUNDS,
-    );
+  public async requestMagicLink(
+    dto: RequestMagicLinkDto,
+  ): Promise<ResponseMagicLinkDto> {
+    try {
+      const token = uuidv4();
+      const nowSeconds = Math.floor(Date.now() / 1000);
+      const expirySeconds =
+        this.configService.get<number>('MAGIC_LINK_EXPIRY_SECONDS') ??
+        magicLinkExpirySeconds;
+      const expiresAt = nowSeconds + expirySeconds;
+
+      await this.dataSource.transaction(async (manager) => {
+        await manager
+          .createQueryBuilder()
+          .delete()
+          .from(MagicLinkTokenEntity)
+          .where('email = :email', { email: dto.email })
+          .execute();
+
+        await manager
+          .createQueryBuilder()
+          .insert()
+          .into(MagicLinkTokenEntity)
+          .values({ token, email: dto.email, expiresAt })
+          .execute();
+      });
 
-    const user = this.userRepo.create({
-      username: loginDto.username,
-      password: hashedPassword,
-    });
+      const backendUrl =
+        this.configService.get<string>('BACKEND_URL') ?? backendUrlEnvKey;
```

Comment:
I didnt get this condition
const backendUrl = this.configService.get<string>('BACKEND_URL')
thats all, if no env - show error